be/src/exprs/jit/jit_engine.cpp

```diff
@@ -123,6 +124,91 @@ StatusOr<JITScalarFunction> JITEngine::compile_scalar_function(ExprContext* cont
     return compiled_function;
 }
 
+Status JITEngine::generate_scalar_function_ir(ExprContext* context, llvm::Module& module, Expr* expr) {
+    llvm::IRBuilder<> b(module.getContext());
+
+    std::vector<Expr*> input_exprs;
+    expr->get_uncompilable_exprs(input_exprs); // duplicated
+    size_t args_size = input_exprs.size();
+
+    /// Create function type.
+    auto* size_type = b.getInt64Ty();
+    // Same with JITColumn.
+    auto* data_type = llvm::StructType::get(b.getInt8PtrTy(), b.getInt8PtrTy());
+    // Same with JITScalarFunction.
+    auto* func_type = llvm::FunctionType::get(b.getVoidTy(), {size_type, data_type->getPointerTo()}, false);
+
+    /// Create function in module.
+    // Pseudo code: void "expr->debug_string()"(int64_t rows_count, JITColumn* columns);
+    auto* func = llvm::Function::Create(func_type, llvm::Function::ExternalLinkage, expr->debug_string(), module);
+    auto* func_args = func->args().begin();
+    llvm::Value* rows_count_arg = func_args++;
+    llvm::Value* columns_arg = func_args++;
+
+    /// Initialize ColumnDataPlaceholder llvm representation of ColumnData
+
+    auto* entry = llvm::BasicBlock::Create(b.getContext(), "entry", func);
+    b.SetInsertPoint(entry);
+
+    // Extract data and null data from function input parameters.
+    std::vector<LLVMColumn> columns(args_size + 1);
+
+    for (size_t i = 0; i < args_size + 1; ++i) {
+        // i == args_size is the result column.
+        auto* jit_column = b.CreateLoad(data_type, b.CreateConstInBoundsGEP1_64(data_type, columns_arg, i));
+
+        const auto& type = i == args_size ? expr->type() : input_exprs[i]->type();
+#if JIT_DEBUG
+        auto tmp = i == args_size ? expr : input_exprs[i];
+        LOG(INFO) << "[JIT] " << i << " col type = " << logical_type_to_string(type.type)
```

Comment:
why not use VLOG，introducing a new macro seems not necessary?

Reply:
is removed in next PR.